routers/common/middleware.go

```diff
@@ -94,6 +138,29 @@ func ForwardedHeadersHandler(limit int, trustedProxies []string) func(h http.Han
 	return proxy.ForwardedHeaders(opt)
 }
 
+// RouteMetrics instruments http requests and responses
+func RouteMetrics() func(h http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
+			inflight := reqInflightGauge.WithLabelValues(req.Method)
+			inflight.Inc()
+			defer inflight.Dec()
+
+			m := httpsnoop.CaptureMetrics(next, resp, req)
+			next.ServeHTTP(resp, req)
+			route := chi.RouteContext(req.Context()).RoutePattern()
+			code := strconv.Itoa(m.Code)
+			reqDurationHistogram.WithLabelValues(req.Method, code, route).Observe(m.Duration.Seconds())
+			respSizeHistogram.WithLabelValues(req.Method, code, route).Observe(float64(m.Written))
```

Comment:
I think we do not need the new vendor package.

We already have `Written Size` support: `services/context/response.go`